lib/model/settings.dart

```diff
@@ -386,4 +436,35 @@ class GlobalSettingsStore extends ChangeNotifier {
     }
     notifyListeners();
   }
+
+  /// The user's choice of the given int-valued setting, or null if not set.
+  ///
+  /// See also [setInt].
+  int? getInt(IntGlobalSetting setting) {
+    return _intData[setting];
+  }
+
+  /// A cache of the [IntGlobalSettings] table in the underlying data store.
+  final Map<IntGlobalSetting, int> _intData;
+
+  /// Set or unset the given int-valued setting,
+  /// persistently for future runs of the app.
+  ///
+  /// A value of null means the setting will be cleared out.
+  ///
+  /// The setting's value will be changed only if new value is different
+  /// from the old one.
+  ///
+  /// See also [getInt].
+  Future<void> setInt(IntGlobalSetting setting, int? value) async {
+    if (value == getInt(setting)) return;
+
+    await _backend.doSetIntGlobalSetting(setting, value);
+    if (value == null) {
+      _intData.remove(setting);
```

Comment:
Similarly let's have this use `_intData` throughout rather than `getInt` (even though in this case it doesn't make a difference in behavior).